### PR TITLE
Fixes #439: Added Cache to Travis Build

### LINKED ISSUE #439: No Cache for Travis Build.
**Current behavior:**
Currently, Travis doesn't have Caching. So that it takes some time for the Test

**Expected behavior:**
Adding Caching to the Travis Build. So that the Test will be done even quicker.

### PR DESCRIPTION
**Pull request by naveentherocker**
Fixes #439 
Added Caching in .travis.yml

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +28,8 @@ before_install:
 install:
 - virtualenv  -p python3 ../flask_env
 - source ../flask_env/bin/activate
-- pip3 install -r requirements/test.txt
+- pip install --upgrade pip
```
Comment (abishekvashok) [inline .travis.yml]: is this required?

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +28,8 @@ before_install:
...
+- pip install -r requirements.txt --cache-dir $HOME/.pip-cache
```
Comment (vaibhavsingh97) [inline .travis.yml]: use `pip3` instead of `pip`

**Comment by naveentherocker** [comment]
Fixed @vaibhavsingh97

**Comment by naveentherocker** [comment]
@vaibhavsingh97 It's better if you could merge it. Please.

File: .travis.yml [hunk 3]
```diff
@@ -25,7 +28,7 @@ before_install:
 install:
 - virtualenv  -p python3 ../flask_env
 - source ../flask_env/bin/activate
-- pip3 install -r requirements/test.txt
+- pip3 install -r requirements.txt --cache-dir $HOME/.pip-cache
```
Comment (abishekvashok) [inline .travis.yml]: requirements/test.txt please

Comment (vaibhavsingh97) [reply .travis.yml]: Nice catch! thanks

**Comment by naveentherocker** [comment]
@abishekvashok I fixed it But Sorry I won't be able to Squash the Commits because I did the changes in Web Editor @vaibhavsingh97

**Comment by abishekvashok** [comment]
You can do that, clone your fork or pull from origin, squash usibg rebase and then force push. However it's not mandatory

**Pull request merged**